src/BlockTypes/ProductCollection.php

```diff
@@ -448,4 +456,86 @@ private function merge_tax_queries( ...$queries ) {
 		// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
 		return [ 'tax_query' => $tax_query ];
 	}
+
+	/**
+	 * Return the `tax_query` for the requested attributes
+	 *
+	 * @param array $attributes  Attributes and their terms.
+	 *
+	 * @return array
+	 */
+	private function get_product_attributes_query( $attributes = array() ) {
+		if ( empty( $attributes ) ) {
+			return array();
+		}
+
+		$grouped_attributes = array_reduce(
+			$attributes,
+			function ( $carry, $item ) {
+				$taxonomy = sanitize_title( $item['taxonomy'] );
+
+				if ( ! key_exists( $taxonomy, $carry ) ) {
+					$carry[ $taxonomy ] = array(
+						'field'    => 'term_id',
+						'operator' => 'IN',
+						'taxonomy' => $taxonomy,
+						'terms'    => array( $item['termId'] ),
+					);
+				} else {
+					$carry[ $taxonomy ]['terms'][] = $item['termId'];
+				}
+
+				return $carry;
+			},
+			array()
+		);
+
+		return array(
+			// phpcs:ignore WordPress.DB.SlowDBQuery
+			'tax_query' => array_values( $grouped_attributes ),
+		);
+	}
+
+	/**
+	 * Return a query to filter products by taxonomies (product categories, product tags, etc.)
+	 *
+	 * For example:
+	 * User could provide "Product Categories" using "Filters" ToolsPanel available in Inspector Controls.
+	 * We use this function to extract it's query from $tax_query.
```

Comment:
I think there is a typo here: "_We use this function to extract **its** query from $tax_query_".

Reply:
I have fixed it in commit [e049cf8](https://github.com/woocommerce/woocommerce-blocks/pull/9634/commits/e049cf891fdda1060f997b83b671bdfe142b4509)